File: trunk/quests/highpass/Captain_Ashlan.pl
#PRIMARY REWARD:  FACTION.  Player turns in one to four orc scalps to Captain Ashlan.

sub EVENT_SAY {
  if($text=~/Hail/i){
    quest::say("Greetings, traveler! I am Captain Ashlan of the Highpass Guards. I keep watch over my men and the [volunteers] here at the East Gate. It's been a busy [job] here lately, with the [orc raids] and all.");
  }
  if($faction>=8||$faction<=5) {
    #[job]
    if($text=~/what job/i){
      quest::say("We're short-handed around here, as usual. Would you like to help us out with the [Volunteer Watch]?");
    }
    #[Volunteer Watch], [Volunteers]
    if($text=~/what volunteer/i){
      quest::say("The Volunteer Watch guards the entry gates of Highpass. Since the [orc raids] are becoming more and more frequent, it's a busy job. But it can pay well, depending on how many [orcs] you slay.");
    }
    #[orcs], [orc raids]
    if($text=~/what orc/i){
      quest::say("One of the orc clans of Kithicor Woods has been trying to expand their territory. Small orc raiding parties are frequently rushing the East Gate. Without the [Volunteer Watch] helping us out, Highpass would probably be overrun by those vile beasts.");
    }
  } else {
    #INCORRECT TEXT
    quest::say("I will not deal with ilk such as you.");
  }
}

sub EVENT_ITEM {
  #Player gives orc scalp
  my $scalps = 0;
  if (plugin::check_handin(\%itemcount,13791=>4)) {$scalps=4;
  } elsif (plugin::check_handin(\%itemcount,13791=>3)) {$scalps=3;
  } elsif (plugin::check_handin(\%itemcount,13791=>2)) {$scalps=2;
  } elsif (plugin::check_handin(\%itemcount,13791=>1)) {$scalps=1;
  }
  if ($scalps>=1) {
    for ($i=0; $i<$scalps; $i++) {
      if ($faction>=8||$faction<=5) {
        my @randomGivenCopper = (0,1,2,3,4,5,6,7,8,9);
        my $copperGiven = ($randomGivenCopper[int(rand(scalar @randomGivenCopper))]);
        my @randomGivenSilver = (0,1,2,3,4,5,6);
        my $silverGiven = ($randomGivenSilver[int(rand(scalar @randomGivenSilver))]);
        my @randomGivenGold = (0,1,2,3);
        my $goldGiven = ($randomGivenGold[int(rand(scalar @randomGivenGold))]);
        my @randomGivenPlatinum = (0,1);
        my $platinumGiven = ($randomGivenPlatinum[int(rand(scalar @randomGivenPlatinum))]);
        quest::say("Great work! Maybe you can help us out again sometime?");
        quest::givecash($copperGiven,$silverGiven,$goldGiven,$platinumGiven);
        quest::faction(149,20);
        quest::faction(31,10);
        quest::faction(214,10);
        quest::faction(53,10);
        quest::faction(105,10);
        quest::exp(10000);
        quest::ding();
      } else {
        #INCORRECT TEXT.  Faction is too low, eat items and laugh.
        quest::say("I will not aid beings such as you.");
      }
    }
  }
  plugin::return_items(\%itemcount); # return unused items
}

#End of File, Zone:highpass  NPC:5013 -- Captain_Ashlan

File: quests/highpass/Captain_Ashlan.lua
function event_say(e)
	local fac = e.other:GetFaction(e.self);
	
	if(e.message:findi("hail")) then
		e.self:Say("Greetings, traveler! I am Captain Ashlan of the Highpass Guards. I keep watch over my men and the [volunteers] here at the East Gate. It's been a busy [job] here lately, with the [orc raids] and all.");
	elseif(fac < 7) then
		if(e.message:findi("job")) then
			e.self:Say("We're short-handed around here, as usual. Would you like to help us out with the [Volunteer Watch]?");
		elseif(e.message:findi("volunteer")) then
			e.self:Say("The Volunteer Watch guards the entry gates of Highpass. Since the [orc raids] are becoming more and more frequent, it's a busy job. But it can pay well, depending on how many [orcs] you slay.");
		elseif(e.message:findi("orc")) then
			e.self:Say("One of the orc clans of Kithicor Woods has been trying to expand their territory. Small orc raiding parties are frequently rushing the East Gate. Without the [Volunteer Watch] helping us out, Highpass would probably be overrun by those vile beasts.");
		end
	else
		e.self:Say("I will not deal with ilk such as you."); -- text made up
	end
end

function event_trade(e)
	local fac = e.other:GetFaction(e.self);
	local scalp = 0;
	local item_lib =require("items");
	
	if(item_lib.check_turn_in(e.trade, {item1 = 13791,item2 = 13791,item3 = 13791,item4 = 13791})) then
			scalp = 4;
	elseif(item_lib.check_turn_in(e.trade, {item1 = 13791,item2 = 13791,item3 = 13791})) then
			scalp = 3;
	elseif(item_lib.check_turn_in(e.trade, {item1 = 13791,item2 = 13791})) then
			scalp = 2;
	elseif(item_lib.check_turn_in(e.trade, {item1 = 13791})) then
			scalp = 1;
	end
	
	if(scalp > 0) then
		if(fac < 7) then
			repeat
				local random_copper = eq.ChooseRandom(0,1,2,3,4,5,6,7,8);
				local random_silver = eq.ChooseRandom(0,1,2,3,4,5,6);
				local random_gold = eq.ChooseRandom(0,1,2,3);
				local random_plat = eq.ChooseRandom(0,1);
				e.self:Say("Great work! Maybe you can help us out again sometime?");
				e.other:Ding();
				e.other:Faction(149,5,0);
				e.other:Faction(31,2,0);
				e.other:Faction(214,2,0);
				e.other:Faction(53,2,0);
				e.other:Faction(105,2,0);
				e.other:AddEXP(1000);
				e.other:GiveCash(random_copper,random_silver,random_gold,random_plat);
				scalp = scalp - 1;
			until scalp == 0
		else
			e.self:Say("I will not aid beings such as you."); -- text made up. faction too low and eat items.
			scalp = 0;
		end
	end
	item_lib.return_items(e.self, e.other, e.trade)
end
